packages/nvidia_nat_core/src/nat/eval/eval_callbacks.py

```
@@ -0,0 +1,96 @@
+                cb.on_eval_complete(result)
+            except Exception:
+                logger.exception("EvalCallback %s.on_eval_complete failed", type(cb).__name__)
+
+    def get_eval_project_name(self) -> str | None:
+        """Get an eval-specific project name from the first callback that supports it."""
+        for cb in self._callbacks:
+            fn = getattr(cb, "get_eval_project_name", None)
+            if fn:
+                try:
+                    return fn()
+                except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)